src/restic/node.go

```diff
@@ -160,32 +160,31 @@ func (node *Node) CreateAt(path string, repo Repository, idx *HardlinkIndex) err
 func (node Node) restoreMetadata(path string) error {
 	var err error
 
-	err = lchown(path, int(node.UID), int(node.GID))
-	if err != nil {
-		return errors.Wrap(err, "Lchown")
-	}
-
 	if node.Type != "symlink" {
-		err = fs.Chmod(path, node.Mode)
-		if err != nil {
+		if err = fs.Chmod(path, node.Mode); err != nil {
 			return errors.Wrap(err, "Chmod")
 		}
 	}
 
 	if node.Type != "dir" {
-		err = node.RestoreTimestamps(path)
-		if err != nil {
+		if err = node.RestoreTimestamps(path); err != nil {
 			debug.Log("error restoring timestamps for dir %v: %v", path, err)
 			return err
 		}
 	}
 
-	err = node.restoreExtendedAttributes(path)
-	if err != nil {
+	if err = node.restoreExtendedAttributes(path); err != nil {
 		debug.Log("error restoring extended attributes for %v: %v", path, err)
 		return err
 	}
 
+	// Do ownership last and only if you are acting as root.
+	if os.Getuid() == 0 {
```

Comment:
That's not a good test, `EUID == 0` is sometimes sufficient, or having `CAP_CHOWN` available. I'd just try it last, and return the error (I have no better idea for it).